graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -542,15 +548,22 @@ def _rename_or_suppress_or_ignore_name_and_add_to_record(
         if (
             new_name_string in self.reverse_name_map
             and self.reverse_name_map[new_name_string] != name_string
-        ):
+        ) or new_name_string in self.custom_scalar_types:
+            # Either new_name_string is the name of a type (that may or may not have been
+            # renamed) in reverse_name_map or new_name_string is the name of a custom scalar.
+            # Either way, we want the original name of the type that is would be named
+            # new_name_string in the renamed schema.
+            other_name_string_renamed_to_new_name_string = self.reverse_name_map.get(
+                new_name_string, new_name_string
+            )
             raise SchemaNameConflictError(
                 '"{}" and "{}" are both renamed to "{}"'.format(
-                    name_string, self.reverse_name_map[new_name_string], new_name_string
+                    name_string, other_name_string_renamed_to_new_name_string, new_name_string
                 )
             )
-        if new_name_string in self.scalar_types or new_name_string in self.builtin_types:
+        if new_name_string in self.custom_scalar_types or new_name_string in builtin_scalar_types:
```

Comment:
seems that the `new_name_string in self.custom_scalar_types` part of the if statement repeated in both this block and the one above? keeping it down here might solve your awkward variable name you mentioned in the comment above as well.

Reply:
Oops! The custom scalar check should only be in the if statement above, since that one checks for renaming conflicts for renamable/ suppressable types, will fix.

That doesn't quite solve the awkward variable name, though-- see follow up comment there.